# Werkcollege-opdrachten Week 1.3

## Dependencies importeren

Kopieer in het codeblok hieronder van het vorige practicum de import-code voor de dependencies die het vaakst worden gebruikt om data in te lezen. Geef er ook de gebruikelijke aliassen aan.<br>
Zet eventuele warnings uit.

In [477]:
import pandas as pd
import sqlite3 as sql
import numpy as np
import warnings
warnings.simplefilter("ignore")

Zet het bestand go_sales_train.sqlite in een makkelijk te vinden map

## Databasetabellen inlezen

Kopieer in het codeblok hieronder van het vorige practicum de code om een connectie met het bestand go_sales_train.sqlite te maken.

In [478]:
sales_conn = sql.connect("go_sales_train.sqlite")

Lees van de ingelezen go_sales_train-database te volgende tabellen in met behulp van "SELECT * FROM *tabel*".
- product
- product_type
- product_line
- sales_staff
- sales_branch
- retailer_site
- country
- order_header
- order_details
- returned_item
- return_reason

In [ ]:
product = pd.read_sql("Select * from product", sales_conn)
product_type = pd.read_sql("Select * from product_type", sales_conn)
product_line = pd.read_sql("Select * from product_line", sales_conn)
sales_staff = pd.read_sql("Select * from sales_staff", sales_conn)
sales_branch = pd.read_sql("Select * from sales_branch", sales_conn)
retailer_site = pd.read_sql("Select * from retailer_site", sales_conn)
country = pd.read_sql("Select * from country", sales_conn)
order_header = pd.read_sql("Select * from order_header", sales_conn)
order_details = pd.read_sql("Select * from order_details", sales_conn)
returned_item = pd.read_sql("Select * from returned_item", sales_conn)
return_reason = pd.read_sql("Select * from return_reason", sales_conn)


Krijg je een "no such table" error? Dan heb je misschien met .connect() per ongeluk een leeg  databasebestand (.sqlite) aangemaakt. <u>Let op:</u> lees eventueel de informatie uit het Notebook van werkcollege 1.1b nog eens goed door.

Als je tijdens onderstaande opdrachten uit het oog verliest welke tabellen er allemaal zijn, kan je deze Pythoncode uitvoeren:

In [480]:
country

COUNTRY_CODE         COUNTRY LANGUAGE CURRENCY_NAME
0              1          France       EN        francs
1              2         Germany       EN         marks
2              3   United States       EN       dollars
3              4          Canada       EN       dollars
4              5         Austria       EN    schillings
5              6           Italy       EN          lira
6              7     Netherlands       EN      guilders
7              8     Switzerland       EN        francs
8              9  United Kingdom       EN        pounds
9             10          Sweden       EN         krona
10            11           Japan       EN           yen
11            12          Taiwan       EN    new dollar
12            13           Korea       EN           won
13            14           China       EN      renminbi
14            16       Australia       EN       dollars
15            17         Belgium       EN        francs
16            18         Denmark       EN        kroner
17            19           Spain       EN       pesetas
18            20          Mexico       EN         pesos
19            21          Brazil       EN         reals
20            22         Finland       EN        markka

erachter 

Let op! Voor alle onderstaande opdrachten mag je <u>alleen Python</u> gebruiken, <u>geen SQL!</u>

## Selecties op één tabel zonder functies

Geef een overzicht met daarin de producten en hun productiekosten waarvan de productiekosten lager dan 100 dollar en hoger dan 50 dollar ligt. (2 kolommen, 23 rijen)

In [481]:
filterd_prodcutionCost = product.loc[(product['PRODUCTION_COST'] > 50) & (product['PRODUCTION_COST'] < 100), ['PRODUCT_NUMBER', "PRODUCTION_COST"]].reset_index(drop=True)


Geef een overzicht met daarin de producten en hun marge waarvan de marge lager dan 20 % of hoger dan 60 % ligt. (2 kolommen, 7 rijen) 

In [482]:
filterd_ProductMarge = product.loc[(product['MARGIN'] < 0.20) | (product['MARGIN'] > 0.60), ['PRODUCT_NUMBER', 'MARGIN']].reset_index(drop=True)


Geef een overzicht met daarin de landen waar met francs wordt betaald. Sorteer de uitkomst op land.  (1 kolom, 3 rijen)

In [483]:
filtered_Currency = country.loc[country['CURRENCY_NAME'] == 'francs', ['COUNTRY', 'CURRENCY_NAME']]


Geef een overzicht met daarin de verschillende introductiedatums waarop producten met meer dan 50% marge worden geïntroduceerd (1 kolom, 7 rijen) 

In [484]:
filterd_IntropductieDatums = product.loc[(product)['MARGIN'] > 0.50, ['INTRODUCTION_DATE']].drop_duplicates()


Geef een overzicht met daarin het eerste adres en de stad van verkoopafdelingen waarvan zowel het tweede adres als de regio bekend is (2 kolommen, 7 rijen)

In [485]:
filterd_VerkoopAfdeling = sales_branch.loc[(sales_branch['ADDRESS2'].notna())  & (sales_branch['REGION'].notna()), ['ADDRESS1', 'CITY']]


Geef een overzicht met daarin de landen waar met dollars (dollars of new dollar) wordt betaald. Sorteer de uitkomst op land. (1 kolom, 4 rijen) 

In [486]:
filterd_Landen = country.loc[(country['CURRENCY_NAME'] == 'dollars') | (country ['CURRENCY_NAME'] == 'new dollar'), ['COUNTRY', 'CURRENCY_NAME']]

Geef een overzicht met daarin beide adressen en de stad van vestigingen van klanten waarvan de postcode begint met een ‘D’ (van duitsland). Filter op vestigingen die een tweede adres hebben. (3 kolommen, 2 rijen) 

In [487]:
filterd_postCode = retailer_site.loc[(retailer_site ['POSTAL_ZONE'].str.startswith('D', na=False)) & (retailer_site['ADDRESS2'].notna()), 
                                     ['ADDRESS1', 'ADDRESS2','CITY', 'POSTAL_ZONE']]

## Selecties op één tabel met functies

Geef het totaal aantal producten dat is teruggebracht (1 waarde) 

In [488]:
TeruggebrachteProducten = returned_item['RETURN_QUANTITY'].sum()
print(TeruggebrachteProducten)

11908


Geef het aantal regio’s waarin verkoopafdelingen gevestigd zijn. (1 waarde)

In [489]:
VerkoopAfdelingRegio = sales_branch['REGION'].nunique()
print(VerkoopAfdelingRegio)

15


Maak 3 variabelen:
- Een met de laagste
- Een met de hoogste
- Een met de gemiddelde (afgerond op 2 decimalen)

marge van producten (3 kolommen, 1 rij) 

In [490]:
LaagsteMarge = product['MARGIN'].min()

HoogsteMarge = product['MARGIN'].max()

GemiddeldeMarge = product['MARGIN'].mean().round(2)

MergingVanMarges = pd.DataFrame({
    'LaagsteMarge': [LaagsteMarge],
    'HoogsteMarge': [HoogsteMarge],
    'GemiddeldeMarge': [GemiddeldeMarge]
})

MergingVanMarges

LaagsteMarge  HoogsteMarge  GemiddeldeMarge
0          0.17           0.7              0.4

Geef het aantal vestigingen van klanten waarvan het 2e adres niet bekend is (1 waarde)

In [491]:
VestigingKlantenZonderTweedeAddress = sales_branch['ADDRESS2'].isna().sum()
print(VestigingKlantenZonderTweedeAddress)

19


Geef de gemiddelde kostprijs van de verkochte producten waarop korting (unit_sale_price < unit_price) is verleend (1 waarde) 

In [492]:
KostprijsWaaropKortingIsVerleend = order_details['UNIT_SALE_PRICE'] < order_details['UNIT_PRICE']
gemiddeldeKostPrijs = order_details.loc[KostprijsWaaropKortingIsVerleend, 'UNIT_PRICE'].mean().round(2)
print(gemiddeldeKostPrijs)

204.99


Geef een overzicht met daarin het aantal medewerkers per medewerkersfunctie (2 kolommen, 7 rijen) 

In [493]:
MedewerkerEnMedewerkerFunctie = sales_staff.groupby('POSITION_EN').size().reset_index(name='SALES_STAFF_CODE')
MedewerkerEnMedewerkerFunctie

POSITION_EN  SALES_STAFF_CODE
0                Branch Manager                19
1              District Manager                 4
2               General Manager                 1
3  Level 1 Sales Representative                13
4  Level 2 Sales Representative                39
5  Level 3 Sales Representative                24
6              Regional Manager                 2

Geef een overzicht met daarin per telefoonnummer het aantal medewerkers dat op dat telefoonnummer bereikbaar is. Toon alleen de telefoonnummer waarop meer dan 4 medewerkers bereikbaar zijn. (2 kolommen, 10 rijen) 

In [494]:
TelefoonNummerVanMedewerkers = sales_staff.groupby('WORK_PHONE').size().reset_index(name='SALE_STAFF_CODE')
TelefoonNummerVanMedewerkers = TelefoonNummerVanMedewerkers[TelefoonNummerVanMedewerkers['SALE_STAFF_CODE'] > 4].reset_index(drop=True)
TelefoonNummerVanMedewerkers

WORK_PHONE  SALE_STAFF_CODE
0    +(44) 121 3505267                5
1  +(61) 03 2982  4242                5
2  +31 (0)20 692 93 94                5
3    +33 1 68 94 52 20                5
4     1 (206) 292-0012                5
5     1 (305) 557-4810                6
6     1 (310) 281-5722                5
7     1 (403) 232-5986                5
8     1 (416) 493-5595                5
9     1 (617) 268-6754                6

## Selecties op meerdere tabellen zonder functies

Geef een overzicht met daarin het eerste adres en de stad van vestigingen van klanten uit ‘Netherlands’ (2 kolommen, 20 rijen) 

In [495]:
retailer_site_Country = retailer_site.merge(country, on='COUNTRY_CODE')
VestigingKlantenUitNetherlands = retailer_site_Country.loc[retailer_site_Country['COUNTRY'] == 'Netherlands', ['ADDRESS1', 'CITY']].reset_index(drop=True)

Geef een overzicht met daarin de productnamen die tot het producttype ‘Eyewear’ behoren. (1 kolom, 5 rijen) 

In [496]:
ProductNamen_ProductType = product.merge(product_type, on='PRODUCT_TYPE_CODE')
EyeWear_ProductType = ProductNamen_ProductType.loc[ProductNamen_ProductType['PRODUCT_TYPE_EN'] == 'Eyewear', ['PRODUCT_TYPE_EN']].reset_index(drop=True)
EyeWear_ProductType

PRODUCT_TYPE_EN
0         Eyewear
1         Eyewear
2         Eyewear
3         Eyewear
4         Eyewear

Geef een overzicht met daarin alle unieke eerste adressen van klantvestigingen en de voornaam en achternaam van de verkopers die ‘Branch Manager’ zijn en aan deze vestigingen hebben verkocht (3 kolommen, 1 rij) 

In [497]:
Sales_staff_sales_branch = sales_staff.merge(sales_branch, on="SALES_BRANCH_CODE")
sales_Staff_Order_header = order_header.merge(Sales_staff_sales_branch, on='SALES_STAFF_CODE')

branch_managers = sales_Staff_Order_header[sales_Staff_Order_header['POSITION_EN'] == 'Branch Manager']

verkopen_per_adres = branch_managers[['ADDRESS1', 'FIRST_NAME', 'LAST_NAME', 'ORDER_NUMBER']].drop_duplicates(subset=['ORDER_NUMBER'])

resultaat = verkopen_per_adres[['ADDRESS1', 'FIRST_NAME', 'LAST_NAME']].drop_duplicates()

resultaat



ADDRESS1 FIRST_NAME LAST_NAME
3185  Avenida Paulista, 333     Bayard     Lopes

Geef een overzicht met daarin van de verkopers hun functie en indien zij iets hebben verkocht de datum waarop de verkoop heeft plaatsgevonden. Laat alleen de verschillende namen van de posities zien van de verkopers die het woord ‘Manager’ in hun positienaam hebben staan. (2 kolommen, 7 rijen) 

In [498]:
order_header_Sales_Staff = sales_staff.merge(order_header, on='SALES_BRANCH_CODE')

Verkoop_Managers = order_header_Sales_Staff[order_header_Sales_Staff['POSITION_EN'].str.contains('Manager', na=False)]

Verkoop_Managers_Met_Verkoop = Verkoop_Managers.dropna(subset=['ORDER_NUMBER'])

Unieke_Posities = Verkoop_Managers_Met_Verkoop.drop_duplicates(subset=['POSITION_EN'])


Unieke_Posities3 = Unieke_Posities.dropna(subset=['ORDER_DATE'])

Unieke_Posities4 = Unieke_Posities3[['POSITION_EN', 'ORDER_DATE']]

Unieke_Posities4


POSITION_EN  ORDER_DATE
0        Branch Manager  2022-04-03
1017   Regional Manager  2022-09-18
2164   District Manager  2022-10-12
17508   General Manager  2022-10-20

Geef een overzicht met daarin de verschillende namen van producten en bijbehorende namen van producttypen van de producten waarvoor ooit meer dan 750 stuks tegelijk verkocht zijn. (2 kolommen, 9 rijen) 

In [499]:
product_Merge_product_Type = product.merge(product_type, on= 'PRODUCT_TYPE_CODE')

product_orders = product_Merge_product_Type.merge(order_details, on='PRODUCT_NUMBER')

product_orders_filtered = product_orders[product_orders['QUANTITY'] > 750]

Overzicht3 = product_orders_filtered[['PRODUCT_NAME', 'PRODUCT_TYPE_EN']].drop_duplicates()


#Overzicht3




Geef een overzicht met daarin de productnamen waarvan ooit meer dan 40% korting is verleend. De formule voor korting is: (unit_price - unit_sale_price) / unit_price (1 kolom, 8 rijen) 

In [500]:
order_details_Product = product.merge(order_details, on='PRODUCT_NUMBER')

order_details_Product['DISCOUNT'] = (order_details_Product['UNIT_PRICE'] - order_details_Product['UNIT_SALE_PRICE']) / order_details_Product['UNIT_PRICE']

Korting_40 = order_details_Product[order_details_Product['DISCOUNT'] > 0.40]

OverzichtKorting = Korting_40[['PRODUCT_NAME']].drop_duplicates()

#OverzichtKorting

Geef een overzicht met daarin de retourreden van producten waarvan ooit meer dan 90% van de aangeschafte hoeveelheid is teruggebracht (return_quantity/quantity). (1 kolom, 3 rijen) 

In [501]:
Order_header_Return_Item = order_details.merge(returned_item, on='ORDER_DETAIL_CODE')
Order_header_Return_Item = Order_header_Return_Item.merge(return_reason, on='RETURN_REASON_CODE')

Order_header_Return_Item['Korting'] = (Order_header_Return_Item['RETURN_QUANTITY'] / Order_header_Return_Item['QUANTITY'])

Korting_90 = Order_header_Return_Item[Order_header_Return_Item['Korting']> 0.90]

ReturnItem = Korting_90[['RETURN_DESCRIPTION_EN']].drop_duplicates()

#ReturnItem

## Selecties op meerdere tabellen met functies

Geef een overzicht met daarin per producttype het aantal producten die tot dat producttype behoren. (2 kolommen, 21 rijen) 

In [502]:
Merge_product_product_type = product.merge(product_type, on='PRODUCT_TYPE_CODE')

Aatal_Producten_per_product_Type = Merge_product_product_type.groupby('PRODUCT_TYPE_EN')["PRODUCT_NUMBER"].count()


Aatal_Producten_per_product_Type = Aatal_Producten_per_product_Type.reset_index(name= 'Aantal Producten')

#Aatal_Producten_per_product_Type



Geef een overzicht met daarin per land het aantal vestigingen van klanten die zich in dat land bevinden. (2 kolommen, 21 rijen) 

In [503]:
Merge_Retailer_Country = retailer_site.merge(country, on='COUNTRY_CODE')

Merge_Retailer_Country = Merge_Retailer_Country.groupby('COUNTRY')['RETAILER_CODE'].count()

Merge_Retailer_Country = Merge_Retailer_Country.reset_index(name= 'Aantal Vestigingen')

#Merge_Retailer_Country

Geef een overzicht met daarin van de producten behorend tot het producttype ‘Cooking Gear’ per productnaam de totaal verkochte hoeveelheid en de gemiddelde verkoopprijs. Sorteer de uitkomst op totaal verkochte hoeveelheid. (4 kolommen, 10 rijen) 

In [504]:
Product_Merge_product_type = product.merge(product_type, on='PRODUCT_TYPE_CODE')
Product_type_product_Merge_Order_detail = Product_Merge_product_type.merge(order_details, on='PRODUCT_NUMBER')

CookingGear = Product_type_product_Merge_Order_detail[Product_type_product_Merge_Order_detail['PRODUCT_TYPE_EN'] == 'Cooking Gear']

ProductEnProductType = CookingGear.groupby('PRODUCT_NAME').agg(
    TOTAL_QUANTITY_VERKOCHT=('QUANTITY', 'sum'),
    AVG_UNIT_PRICE=('UNIT_PRICE', 'mean'),
    TOTAL_ORDER=('ORDER_NUMBER', 'count')
).reset_index()


#ProductEnProductType

Geef een overzicht met daarin per land de naam van het land, de naam van de stad waar de verkoopafdeling is gevestigd (noem de kolomnaam in het overzicht ‘verkoper’) en het aantal steden waar zich klanten bevinden in dat land (noem de kolomnaam in het overzicht ‘klanten’) (3 kolommen, 29 rijen) 

In [505]:
SalesBranch_Country = sales_branch.merge(country, on='COUNTRY_CODE')
SalesBranch_Country_SalesStaff = SalesBranch_Country.merge(sales_staff, on='SALES_BRANCH_CODE')

SalesBranch_Country_SalesStaff = SalesBranch_Country.groupby('CITY')['COUNTRY_CODE'].count().reset_index()
SalesBranch_Country_SalesStaff.rename(columns={'CITY': 'klanten', 'COUNTRY_CODE': 'Aantal_verkoopafdelingen'}, inplace=True)


#SalesBranch_Country_SalesStaff



## Pythonvertalingen van SUBSELECT en UNION met o.a. for-loops

Geef een overzicht met daarin de voornaam en de achternaam van de medewerkers die nog nooit wat hebben verkocht (2 kolommen, 25 rijen) 

In [506]:
#OrderHeader_SalesStaff = sales_staff.merge(order_header, on='SALES_STAFF_CODE')
#Verkoop_Medewerkers = order_header['SALES_STAFF_CODE'].unique()
#NietVerkocht = sales_staff[~sales_staff['SALES_STAFF_CODE'].isin(Verkoop_Medewerkers)]
#resultaat = NietVerkocht[['FIRST_NAME','LAST_NAME']].dropna()
#resultaat

#merged = sales_staff.merge(order_header, on='SALES_STAFF_CODE', how='left')
#niet_Verkocht = merged[merged['ORDER_NUMBER'].isna()][['FIRST_NAME', 'LAST_NAME']].reset_index()
#niet_Verkocht

Verkoop_Medewerkers = set(order_header['SALES_STAFF_CODE'])
Niet_Verkocht = []
for _, row in sales_staff.iterrows():
    if row['SALES_STAFF_CODE'] not in Verkoop_Medewerkers: 
        Niet_Verkocht.append((row ['FIRST_NAME'], row['LAST_NAME']))

df_Niet_Verkocht = pd.DataFrame(Niet_Verkocht, columns=['FIRST_NAME', 'LAST_NAME'])
#df_Niet_Verkocht


Geef een overzicht met daarin het aantal producten waarvan de marge lager is dan de gemiddelde marge van alle producten samen. Geef in het overzicht tevens aan wat de gemiddelde marge is van dit aantal producten waarvan de marge lager dan de gemiddelde marge van alle producten samen is. (1 kolom, 2 rijen) 

In [507]:
Gemiddlede_Marge = product['MARGIN'].mean()

Lagere_MARGE_PRODUCTEN = product[product['MARGIN'] < Gemiddlede_Marge]


aantal_producten = len(Lagere_MARGE_PRODUCTEN)

Gemiddlede_Marge = Lagere_MARGE_PRODUCTEN['MARGIN'].mean()

Resultaal22 = pd.DataFrame({
    'Waarde' : [aantal_producten, Gemiddlede_Marge]
}, index=['Aantal Producten', 'Gemiddelde marge'])
#Resultaal22


Geef een overzicht met daarin de namen van de producten die voor meer dan 500 (verkoopprijs) zijn verkocht maar nooit zijn teruggebracht. (1 kolom, 13 rijen) 

In [508]:
#Merged_Orderdetails = order_details.merge(returned_item, on='ORDER_DETAIL_CODE')
#Merged_Orderdetails2 = Merged_Orderdetails.merge(product, on='PRODUCT_NUMBER')
#Producten500 = Merged_Orderdetails2[Merged_Orderdetails2['UNIT_SALE_PRICE'] > 500]
#ItemsNotReturned = Producten500[Producten500['ORDER_DETAIL_CODE'].isna()]
#Resultaat333 = ItemsNotReturned[['PRODUCT_NAME', ]]
#Resultaat333

Product500 = order_details[order_details['UNIT_SALE_PRICE'] > 500]
Gererourneed = set(returned_item['ORDER_DETAIL_CODE'])
ItemsNotReturned = Product500[~Product500['ORDER_DETAIL_CODE'].isin(Gererourneed)]
Resultaat44 = ItemsNotReturned[['PRODUCT_NUMBER','UNIT_SALE_PRICE']].drop_duplicates()
Resultaat44








PRODUCT_NUMBER  UNIT_SALE_PRICE
88                 12           650.89
89                 12           573.88
91                 12           620.21
92                 12           637.61
95                 13           532.26
...               ...              ...
37738             102           590.46
37742             104           516.65
37743             105           889.45
37745             107           814.53
37747             108           661.09

[1071 rows x 2 columns]

Geef een overzicht met daarin per (achternaam van) medewerker of hij/zij manager is of niet, door deze informatie toe te voegen als extra 'Ja/Nee'-kolom.<br>
Hint: gebruik een for-loop waarin je o.a. bepaalt of het woord 'Manager' in de functie (position_en) staat. (2 kolommen, 102 rijen).

In [509]:
sales_staff = pd.DataFrame(sales_staff)

# Voeg de kolom 'is_manager' toe
sales_staff['is_manager'] = sales_staff['POSITION_EN'].apply(lambda x: 'Ja' if 'Manager' in x else 'Nee')

# Reset de index en voeg de oude index toe als een kolom
sales_staff_reset = sales_staff.reset_index()

# Maak een nieuwe DataFrame met alleen de gewenste kolommen
result_df = sales_staff_reset[['index', 'FIRST_NAME', 'is_manager']]

# Print de nieuwe DataFrame
result_df

index FIRST_NAME is_manager
0        0      Denis         Ja
1        1  Élizabeth        Nee
2        2      Émile        Nee
3        3    Étienne        Nee
4        4    Elsbeth        Nee
..     ...        ...        ...
97      97      Giele        Nee
98      98   François        Nee
99      99     Yvette        Nee
100    100      Willi        Nee
101    101     Sabine        Nee

[102 rows x 3 columns]

Met de onderstaande code laat je Python het huidige jaar uitrekenen.

In [510]:
from datetime import date
date.today().year

2025

Met de onderstaande code selecteer je op een bepaald jaartal uit een datum.

In [511]:
from datetime import datetime

date_str = '16-8-2013'
date_format = '%d-%m-%Y'
date_obj = datetime.strptime(date_str, date_format)

date_obj.year

2013

Geef met behulp van bovenstaande hulpcode een overzicht met daarin op basis van het aantal jaar dat iemand in dienst is of een medewerker ‘kort in dienst’ (minder dan 25 jaar in dienst) of een ‘lang in dienst’ (groter gelijk dan 12 jaar in dienst) is. Geef daarbij per medewerker in een aparte kolom zowel ‘kort in dienst’ als ‘lang in dienst’ aan. Gebruik (wederom) een for-loop.<br>
(2 kolommen, 102 rijen) 

In [522]:
import pandas as pd
from datetime import datetime

# Aangenomen dat sales_staff al een dataset is met een 'DATE_HIRED' kolom
sales = pd.DataFrame(sales_staff)

# Functie om het aantal jaren in dienst te berekenen
def jaren_in_dienst(DATE_HIRED):
    DATE_HIRED_obj = datetime.strptime(DATE_HIRED, '%Y-%m-%d')
    huidig_jaar = datetime.today().year
    return huidig_jaar - DATE_HIRED_obj.year

# Bereken jaren in dienst en voeg deze toe als een nieuwe kolom
sales['JAREN IN DIENDS'] = sales['DATE_HIRED'].apply(jaren_in_dienst)

# Maak kolommen 'Korte In Dienst' en 'Lange In Dienst' op basis van de dienstjaren
sales['Korte In Dienst'] = sales['JAREN IN DIENDS'].apply(lambda x: 'Ja' if x < 11 else 'Nee')
sales['Lange In Dienst'] = sales['JAREN IN DIENDS'].apply(lambda x: 'Ja' if x >= 12 else 'Nee')

# Filteren van medewerkers met zowel 'Korte In Dienst' als 'Lange In Dienst' gelijk aan 'Ja'
sales2 = sales[(sales['Korte In Dienst'] == 'Ja') | (sales['Lange In Dienst'] == 'Ja')].reset_index()

sales2 = sales2[['Korte In Dienst', 'Lange In Dienst']]

# Toon het resultaat
sales2


Korte In Dienst Lange In Dienst
0               Nee              Ja
1               Nee              Ja
2               Nee              Ja
3               Nee              Ja
4               Nee              Ja
..              ...             ...
97              Nee              Ja
98              Nee              Ja
99              Nee              Ja
100             Nee              Ja
101             Nee              Ja

[102 rows x 2 columns]

## Van Jupyter Notebook naar Pythonproject

1. Richt de map waarin jullie tot nu toe hebben gewerkt in volgens de mappenstructuur uit de slides.
2. Maak van de ontstane mappenstructuur een Pythonproject dat uitvoerbaar is vanuit de terminal. Maak daarin een .py-bestand dat minstens 5 antwoorden uit dit notebook (in de vorm van een DataFrame) exporteert naar Excelbestanden. Alle notebooks mogen als notebook blijven bestaan.
3. Zorg ervoor dat dit Pythonproject zijn eigen repo heeft op Github. Let op: je virtual environment moet <b><u>niet</u></b> meegaan naar Github.

Je mag tijdens dit proces je uit stap 1 ontstane mappenstructuur aanpassen, zolang je bij het beoordelingsmoment kan verantwoorden wat de motivatie hierachter is. De slides verplichten je dus nergens toe.